lfc/core/src/main/kotlin/org/lflang/generator/uc/UcConnectionGenerator.kt

```diff
@@ -44,27 +44,35 @@ class UcGroupedConnection(val varRef: VarRef, val conn: Connection, val uid: Int
 class UcChannel(val varRef: VarRef, val port_idx: Int, val bank_idx: Int) {
     private val portOfContainedReactor = varRef.container != null
     private val index = "${if (portOfContainedReactor) "[${bank_idx}]" else ""}[${port_idx}]"
-
     private val reactorInstance = if (portOfContainedReactor) "${varRef.container.name}[${bank_idx}]." else ""
     private val portInstance = "${varRef.name}[${port_idx}]"
 
     fun generateChannelPointer() = "&self->${reactorInstance}${portInstance}"
 }
 
-
 // Wrapper around a variable reference to a Port. Contains a channel for each bank/multiport within it.
 // For each connection statement where a port is referenced, we create an UcPort and use this class
 // to figure out how the individual channels are connect to other UcPorts.
 class UcPort(val varRef: VarRef) {
     val bankWidth = varRef.container?.width?:1
     val portWidth = (varRef.variable as Port).width
+    private val isInterleaved = varRef.isInterleaved
     private val channels = ArrayDeque<UcChannel>()
 
-    // Construct the stack of channels belonging to this port.
+    // Construct the stack of channels belonging to this port. If this port is interleaved,
+    // then we create channels first for ports then for banks.
     init {
-        for (i in 0..bankWidth-1) {
-            for (j in 0..portWidth-1) {
-                channels.add(UcChannel(varRef, j, i))
+        if (isInterleaved) {
```

Comment:
Just to put the brain into your head: 

https://github.com/lf-lang/lingua-franca/blob/master/core/src/main/resources/lib/cpp/lfutil.hh#L69